### ISSUE TITLE
pypyr and Win 10

### ISSUE THREAD

**Issue by kontramind** [pypyr/pypyr#248]
Q; Has pypyr been tested on Windows?

Many thanks
Aleksandar

**Comment by yaythomas** [comment]
Hi Aleksandar, thanks for your interest in pypyr!

Yes, pypyr should absolutely work on Windows! I know the documentation tend to give its examples using POSIX paths, but I'm careful to keep all code cross-platform compatible, some contributors are on Windows, and I know of some happy Windows users too!

**Comment by kontramind** [comment]
@yaythomas Dear Thomas, many thanks for quick reply.
This fantastic project seem to be a good fit for our rendering workflow.
I have just started to experiment with it, However, I'm not able to successfully execute the following pipeline

`- name: pypyr.steps.shell
  in:
    cmd: 'echo hello'
`

I'm on Win 10, Python 3.9.8 and getting the following error:

Something went wrong. Will now try to run on_failure
←[91mAttributeError: 'CommentedSeq' object has no attribute 'get'←[0;0m

Please can you tell what I'm doing wrong.

**Comment by kontramind** [comment]
@yaythomas This seems to be connected to cmd/shell contexts

**Comment by yaythomas** [comment]
You left out the default `steps` groups (the entry-point - sort of the equivalent of a "main()") - so your pipeline should look like this:

```yaml
steps:
  - name: pypyr.steps.shell
    in:
      cmd: 'echo hello'
```

But that error message is really not great, thank you for bringing it to my attention! pypyr should do a better job to make a less mysterious more user-friendly msg in this case where the pipeline yaml is unexpected!

**Comment by kontramind** [comment]
@yaythomas  Yes it worked now. Sorry for my noob question (feeling a bit ashamed now).
Do you want me to close the issue ir leave it open for some time as I might hit some other issues?

**Comment by yaythomas** [comment]
No problem @kontramind, I actually feel ashamed that the error msg wasn't more useful! Throughout the rest of pypyr in general (I think) error messages are pretty descriptive and useful, and the steps etc. normally go to a fair bit of effort to validate inputs to provide user-friendly messages on unexpected inputs.

I have already opened an issue referencing this one to keep track of this. I might not deploy this fix on its own because it's maybe a little small, but it'll be in the next release :-)

You can happily close this issue, if you don't mind. If you have other questions, you can feel free to use the Discussion board for questions/comments/ideas https://github.com/pypyr/pypyr/discussions, and of course, if you do hit Issues please feel free to log them too!

**Comment by kontramind** [comment]
Many thanks. Closing the issue/question.

**Comment by yaythomas** [comment]
Thanks Aleksandar :-)

FYI, fix coded+tested & merged into main, see #249. This will be in the next release.

In future, err msg on this sort of err will be:
```
PipelineDefinitionError: A pipeline must be a mapping at the top level. Does your top-level yaml have a 'steps:' key? For example:

steps:
  - name: pypyr.steps.echo
    in:
      echoMe: this is a bare bones pipeline example.
```

**Comment by kontramind** [comment]
@yaythomas Wonderful. Thanks.
pypyr is really amazing as I get to know it more and more,
